### PR TITLE
docs: update register preview template parameter description

### LINKED ISSUE #1197: Preview Template for File Collection
**- Do you want to request a *feature* or report a *bug*?**

Bug

**- What is the current behavior?**

When I use `CMS.registerPreviewTemplate` for my file collection it doesn't work.

**- If the current behavior is a bug, please provide the steps to reproduce.**

Create a file collection.
Run `registerPreviewTemplate`

**- What is the expected behavior?**

I would expect to get my preview template rendered so that I can show previews for this content.

**- Please mention your CMS, node.js, and operating system version.**

Netlify CMS version 1.3.5
Node v8.9.4
OS X 10.13.3

**- Please link or paste your `config.yml` below if applicable.**
Excerpt:
```
collections:
  - name: "pages"
    label: "Pages"
    files:
      - name: "about-us"
        label: "About Us Page"
        description: "Content for the About Us Page"
        file: "content/about-us-page.json"
        fields:
          - {label: "Heading", name: "heading", widget: "string"}
          - {label: "Body", name: "body_markdown", widget: "markdown"}
```

Then:
```javascript
CMS.registerPreviewTemplate('pages', PagesPreviewTemplate);
```

Any help anyone can provide would be greatly appreciated.

### PR DESCRIPTION
**Pull request by runandrew**
**Summary**

This feature branch updates the API documentation for `CMS.registerPreviewTemplate`. The issue that gave motivation for this PR is #1625. The goal is to make the documentation more clear for registering file collection templates.

This also closes #1197.

### CODE REVIEW

**Comment by verythorough** [comment]
Deploy preview for *netlify-cms-www* ready!

Built with commit 5a9153e969f1c4053bc91ff27e72b1225a2bc9f8

https://deploy-preview-1667--netlify-cms-www.netlify.com

**Comment by verythorough** [comment]
Deploy preview for *cms-demo* ready!

Built with commit 5a9153e969f1c4053bc91ff27e72b1225a2bc9f8

https://deploy-preview-1667--cms-demo.netlify.com

File: website/content/docs/customization.md [hunk 1]
```diff
@@ -54,13 +54,15 @@ body {
...
-Registers a template for a collection.
+Registers a template for a folder collection or an individual file in a file collection.
 
-`CMS.registerPreviewTemplate(collection, react_component);`
+`CMS.registerPreviewTemplate(name, react_component);`
 
 **Params:**
 
-* collection: The name of the collection which this preview component will be used for.
+* name: The name of the entry which this preview component will be used for.
```
Comment (tech4him1) [inline website/content/docs/customization.md]: > The name of the entry ...

I'm not so sure we should use the word "entry" here, as that is typically used to refer to a specific "file" in the docs. Any other ideas?

**Comment by runandrew** [comment]
I was having trouble with this and at the last minute changed it to “entry”. I had also thought about using “resource” but I thought that might be too formal or too vague. 

I was hoping the sub bullet points would provide enough context of what it is referring to.

**Comment by erquhart** [comment]
@runandrew I'd suggest rewording to something like:

`name: the name of the collection (or file for file collections) which this preview component will be used for`

That together with your bullets should be pretty clear.

File: website/content/docs/customization.md [hunk 2]
```diff
@@ -54,13 +54,15 @@ body {
...
 
-`CMS.registerPreviewTemplate(collection, react_component);`
+`CMS.registerPreviewTemplate(name, react_component);`
 
 **Params:**
 
-* collection: The name of the collection which this preview component will be used for.
+* name: The name of the collection (or file for file collections) which this preview component will be used for.
+  * Folder collections: Use the name of the collection
+  * File collections: Use the name of the file
```
Comment (tech4him1) [inline website/content/docs/customization.md]: Let's clarify what the "name" of the file is. I believe it is the slug (without file extension or path), correct?

Comment (runandrew) [reply website/content/docs/customization.md]: In the code I see it's `slug` for File Collections:
https://github.com/netlify/netlify-cms/blob/f801b192211b1e1953ced969a3093eab9e04e14d/packages/netlify-cms-core/src/reducers/collections.js#L94-L96

For Folder Collections it's the `name`:
https://github.com/netlify/netlify-cms/blob/f801b192211b1e1953ced969a3093eab9e04e14d/packages/netlify-cms-core/src/reducers/collections.js#L61-L63

But, in the admin config, it's not necessary to provide a `slug` for files:
https://github.com/netlify/netlify-cms/blob/f801b192211b1e1953ced969a3093eab9e04e14d/dev-test/config.yml#L55-L66

It seems like the `slug` ends up being the `name` for a file in a file collection, but I'm not sure where that is declared.

Comment (runandrew) [reply website/content/docs/customization.md]: I think I found where the `slug` is determined:
https://github.com/netlify/netlify-cms/blob/f801b192211b1e1953ced969a3093eab9e04e14d/packages/netlify-cms-core/src/reducers/collections.js#L78-L84

With that, do you think the current wording is okay? Or explicitly say the `name` field of the file in a file collection or folder collection.

Comment (tech4him1) [reply website/content/docs/customization.md]: I think that will be fine, sorry!
